code/modules/mob/mob.dm

```diff
@@ -600,21 +600,21 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 
 	if(next_move >= world.time)
 		return
-	if(!src || !isturf(src.loc))
-		return 0
-	if(istype(A, /obj/effect/temp_visual/point))
-		return 0
+	if(!isturf(loc) || istype(A, /obj/effect/temp_visual/point))
+		return FALSE
 
 	var/tile = get_turf(A)
 	if(!tile)
-		return 0
+		return FALSE
 
 	changeNext_move(CLICK_CD_POINT)
 	var/obj/P = new /obj/effect/temp_visual/point(tile)
 	P.invisibility = invisibility
-	P.pixel_x = A.pixel_x
-	P.pixel_y = A.pixel_y
-	return 1
+	// Start off from the pointer and make it slide to the pointee
+	P.pixel_x = (x - A.x) * 32
+	P.pixel_y = (y - A.y) * 32
```

Comment:
Questioning why you dont just use the mobs pixel_x values

Reply:
Because `pixel_x` and `pixel_y` are relative values starting from the atom's turf. Here's an image that (hopefully) explains the math better:
![image](https://user-images.githubusercontent.com/1496804/100093035-4172b100-2e57-11eb-8752-45c2b8077848.png)